### PR TITLE
Aarondonahue/p3 ice supsat fix

### PR DESCRIPTION
**Pull request by AaronDonahue**
Brings scream-master up to date with bugfixes that were merged into dyamond2 (see #714).

Addresses issue #861 

[NBFB]

### CODE REVIEW

**Comment by AaronDonahue** [comment]
We would expect this test to fail `p3_run_and_cmp` since it is non-BFB in P3.

**Comment by AaronDonahue** [comment]
@PeterCaldwell , I think this PR needs to be re-reviewed before it can be merged.

**Review by PeterCaldwell — APPROVED** [review]
Thanks Aaron!

**Comment by bartgol** [comment]
@AaronDonahue I removed the automerge label, cause the AT sees all checks passing, and tries to merge (even though there are conflicts with master). You will have to merge master and fix conflicts. Then the AT will test again the branch.

**Comment by AaronDonahue** [comment]
@luca, ok great.  Thanks for the heads up.  I'll do that now.

**Comment by bartgol** [comment]
> @luca, ok great. Thanks for the heads up. I'll do that now.

Wrong Luca. =)

**Comment by PeterCaldwell** [comment]
@bartgol - so is it our new official policy to not use the automerge feature on any of our PRs? This might be worth discussing on Thursday...

**Comment by AaronDonahue** [comment]
> > @luca, ok great. Thanks for the heads up. I'll do that now.
> 
> Wrong Luca. =)

Ha! you would think that git would have figured out who I like to tag more often...

**Comment by bartgol** [comment]
> @bartgol - so is it our new official policy to not use the automerge feature on any of our PRs? This might be worth discussing on Thursday...

No, automerge is fine. The issue is that if the reviewer takes "too long" to reiview (sometimes that can be 1h, sometimes 1 week, depending on how many other PRs go in that touch similar paths), master can change while the PR is still open.

However, the AT sees that the tests passed, and no other commit was pushed _on this branch_, so it _assumes_ the branch is still ok. Once the reviewer approves, it then proceeds to automerge. I asked to the AT devs if the AT could at least check the PR mergeability (what we see at the bottom) before attempting to merge. That would prevent the AT from crashing. A much better (but probably more involved to implement) approach would be for the AT to store the sha of the master branch when tests pass, and deem the check mark as 'stale' if the sha of master changes (hence triggering a new retest automatically).

*tl;dr* No, we still want to keep automerge, cause "usually" it works. Sometimes they don't, like in this case, but it's not _that_ common.

**Comment by AaronDonahue** [comment]
I'm going to merge this now.  It passed the AT and would just need to be re-reviewed.  But I haven't changed anything except merging in master.

**Review by PeterCaldwell — APPROVED** [review]
